test/FunctionalTests/Microsoft.OData.Core.Tests/Json/ODataUtf8JsonWriterAsyncTests.cs

```diff
@@ -401,8 +435,7 @@ public async Task WriteODataValueAsync_WritesODataCollectionValue()
                 new object[] { Encoding.UTF8 },
                 new object[] { Encoding.Unicode },
                 new object[] { Encoding.UTF32 },
-                new object[] { Encoding.BigEndianUnicode },
-                new object[] { Encoding.ASCII }

```

Comment:
Why remove `Encoding.ASCII`?

Reply:
Good catch. Had forgotten to mention that. The reason I removed `Encoding.ASCII` is because when we don't escape these non-ascii characters, their binary representation is outside the ASCII range, so the output text contains characters that ASCII can't represent and end up getting rendered with placeholders like `?`. Technically the writer still supports ascii so long as the input is ascii (since non-ascii characters are not automatically escaped).

I have added a new test that verifies ASCII support when `JavaScriptEncoder.Default` is used (which escapes non-ascii characters).